test/spmd/test_xla_sharding.py

```diff
@@ -1151,6 +1151,55 @@ def test_spmd_full_to_shard_shape(self):
         f'custom_call_target="SPMDFullToShardShape", sharding={{manual}}', hlo)
     self.assertEqual(torch_xla._XLAC._get_xla_sharding_spec(xx), "{manual}")
 
+  def test_spmd_shard_to_full_shape(self):
+    x = torch.zeros(8, 8).to(xm.xla_device())
+    x += 1
+    # No sharding spec attached.
+    with self.assertRaises(RuntimeError):
+      x = torch_xla._XLAC._spmd_shard_to_full_shape(
+          x, torch_xla._XLAC.OpSharding([], [], [], xs.ShardingType.REPLICATED),
+          x.shape, x.dtype)
```

Comment:
so we've decided to pass the original full shape as argument, right?

Reply:
At least I have decided haha, see the PR descriptions for the reasons behind.